SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/Base.h

```diff
@@ -411,59 +412,55 @@ class SOFA_CORE_API Base
     /// \code  T* ptr = nullptr; std::string type = T::typeName(ptr); \endcode
     /// This way derived classes can redefine the typeName method
     template<class T>
-    static std::string typeName(const T* ptr= nullptr)
+    static std::string typeName()
     {
-        return BaseClass::defaultTypeName(ptr);
+        return T::GetClass()->typeName;
     }
 
     /// Helper method to get the class name of a type derived from this class
     ///
     /// This method should be used as follow :
-    /// \code  T* ptr = nullptr; std::string type = T::className(ptr); \endcode
+    /// \code  std::string type = Base::className<B>(); \endcode
     /// This way derived classes can redefine the className method
     template<class T>
-    static std::string className(const T* ptr= nullptr)
```

Comment:
Do we rename that to ClassName() ? 
- tag the deprecated
- refait une ClassName()